src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -19,92 +19,135 @@
         string branchString = context.Request.QueryString["branch"];
         string prString = context.Request.QueryString["pr"];
 
+        uint repoId;
+        if (string.IsNullOrWhiteSpace(repoString) || !uint.TryParse(repoString, out repoId))
+        {
+            context.Response.StatusCode = 400;
+            context.Response.Write(string.Format("The repo ID specified, '{0}', is invalid <br/>", repoString ?? "unspecified"));
+            return;
+        }
+
         if (string.IsNullOrWhiteSpace(prString) && string.IsNullOrWhiteSpace(branchString))
         {
             context.Response.StatusCode = 400;
-            context.Response.Write("No PR or branch specified");
+            context.Response.Write("No PR or branch specified. Specify either a PR or branch <br/>");
             return;
         }
 
         if (!string.IsNullOrWhiteSpace(prString) && !string.IsNullOrWhiteSpace(branchString))
         {
             context.Response.StatusCode = 400;
-            context.Response.Write("PR and branch specified; please specify only one of them to sync to <br/>");
+            context.Response.Write("Both PR and branch specified; please specify only one of them to sync to <br/>");
+            return;
+        }
+
+        context.Response.Write(string.Format("PR = {0}<br/>", HttpUtility.HtmlEncode(prString) ?? "unspecified"));
+        context.Response.Write(string.Format("branch = {0}<br/>", HttpUtility.HtmlEncode(branchString) ?? "unspecified"));
+
+        // Check for prerequisites needed before this script can execute
+        string gitRepoPath = GetRepoPath(repoId);
+        success = CheckPrerequisites(gitRepoPath, result);
+
+        if (!success)
+        {
+            // CheckPrerequisites failed, this is a server issue
+            context.Response.StatusCode = 500;
+            context.Response.Write(HttpUtility.HtmlEncode(result.ToString()));
             return;
         }
 
-        context.Response.Write(string.Format("PR = {0}<br/>", HttpUtility.HtmlEncode(prString) ?? "Unspecified"));
-        context.Response.Write(string.Format("branch = {0}<br/>", HttpUtility.HtmlEncode(branchString) ?? "Unspecified"));
+        // Cleanup all branches except master from the repo - this shouldn't fail, and if it does, there's something wrong with the repo
+        success = CleanupBranches(gitRepoPath, result);
+
+        if (!success)
+        {
+            // CleanupMergedBranches failed, this is a server issue
+            context.Response.StatusCode = 500;
+            context.Response.Write(HttpUtility.HtmlEncode(result.ToString()));
+            return;
+        }
 
         // Do stuff
+        success = false;
 
         uint pr;
         if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
         {
-            success = SyncToPr(pr, result);
+            success = SyncToPr(pr, gitRepoPath, result);
         }
         else if (!string.IsNullOrEmpty(branchString))
```

Comment:
Since you have already validated all input parameters, you can change code to below to be simpler. Less code, happier developer 😄 
```
if (!string.IsNullOrEmpty(prString) {
    if (uint.TryParse(prString, out pr))  {
        success = syncToPr(pr, gitRepoPath, result);
    }
}
else {
 ...
}
```